functions/src/Services/Event.ts

```diff
@@ -35,30 +46,40 @@ export const rsvp = (event: Event, user: User) => {
 	// App.database().ref(`events/${event.id}/rsvp`).update({ [user.id]: !(user.id in event.rsvp) });
 };
 
+/**
+ * userReference and eventReference do not use their respective converters, because they're stored as documentReferences in the firestore
+ *
+ * The attendees subCollection inside of an event stores documentReferences to users that attended that event,
+ * that way if any of the user data updates, we don't have to update the subCollection inside of an event
+ *
+ * The eventsAttended subCollection works the same way as the attendees subCollection, except this one stores references to events attended by the user
```

Comment:
After thinking about it a little I believe these comments should actually be in the user model and the events model. The comments in check in should talk about particularities in why you did things the way you did. for example why do you have two thens, like why is that necessary as opposed to doing it in 1 call(and if maybe storing the data in a different way would allow you to do it in 1 call then why didn't we do it that way.)

Reply:
tldr: Work these comments into Event and User Model and explain why you chose the implementation of check in that you did.

Reply:
The thing is, that these collections aren't created until checkIn event is run, so the models for user and event don't have the subcollections, so then wouldn't it be redundant to add the comment to the model?

Also the 2 thens are there more for style? should I do it in 1 then instead?

Reply:
@BautistaBertolami I still think the implementation of how they're represented should be in the Model. I think the converter would be in charge of converting from the firebase reference stuff to the model